src/sql/azdata.proposed.d.ts

```diff
@@ -823,11 +823,61 @@ declare module 'azdata' {
 		parentTypeName?: string;
 	}
 
+	/**
+	 * Represents a selected range in the result grid.
+	 */
+	export interface SelectionRange {
+		fromRow: number;
+		toRow: number;
+		fromCell: number;
+		toCell: number;
+	}
+
+	/**
+	 * Parameters for the copy results request.
+	 */
+	export interface CopyResultsRequestParams {
+		/**
+		 * ownerUri of the editor
+		 */
+		ownerUri: string;
+		/**
+		 * Index of the batch.
+		 */
+		batchIndex: number;
```

Comment:
Maybe make this optional? Not all providers might have the concept of batches...

Reply:
The properties to identify a result are the same as existing subset query request parameters: https://github.com/microsoft/azuredatastudio/blob/55cedb810dc8783d68ccabb2021e6abe35ce74d6/src/sql/azdata.d.ts#L1180.

other providers can handle this properly.